# Relatório de Teste A/B

Nesse relatório, nosso grupo foi além devido a implementação do teste de Fisher, que pode ser visualizado na sessão 6, denominada "Testes estatísticos".

### Sumário

1. **Introdução**
   - Apresentação do objetivo do relatório e das estratégias propostas para otimizar a taxa de conversão e a qualidade dos leads do WizardOn.

2. **Objetivo do Teste**
   - Avaliar se a alteração na localização do formulário na página aumenta a taxa de conversão de leads de 5% para 7%.

3. **Método**
   - Descrição do design do teste A/B, incluindo a definição dos grupos de controle e experimental, as métricas utilizadas e as hipóteses testadas.

4. **Teste A/B**
   - Detalhamento das alterações realizadas na página e da metodologia de implementação do teste.

5. **Preparação dos Dados**
   - Descrição dos dados coletados, processos de carregamento e unificação dos dados, e cálculo da taxa de conversão.

6. **Descrição dos Dados**
   - Estatísticas descritivas das sessões, conversões e taxa de conversão.

7. **Testes Estatísticos**
   - Apresentação e interpretação dos resultados dos testes t, intervalos de confiança e Teste de Fisher.

8. **Análise dos Resultados**
   - Visualização gráfica dos resultados e interpretação detalhada dos dados obtidos.

9. **Limitações**
   - Discussão das limitações do teste realizado, como período curto, amostra pequena e múltiplas mudanças simultâneas.

10. **Conclusão**
    - Resumo dos achados e recomendações para futuras otimizações da página com base nos resultados obtidos.


## 1. Introdução

O documento propõe uma solução focada na otimização da comunicação da modalidade de ensino WizardOn, na estabilização e aumento da taxa de conversão de visitas para leads e na melhoria da qualidade dos leads capturados para a Wizard by Pearson. Isso será alcançado através da análise dos problemas identificados e da implementação de novas estratégias.

### 1.1. Objetivo do teste

O objetivo deste teste A/B é avaliar se a alteração na localização do formulário na página aumenta a taxa de conversão de leads de 5% para 7%, em comparação com a versão atual.

## 2. Método

### 2.1. Design do Teste

O teste será realizado com duas páginas: a página de controle será a própria página da Wizard com a campanha do Dia dos Namorados. A página do grupo experimental terá duas sessões: uma com a campanha do dia dos namorados e outra com uma alteração, que consiste em mover o formulário de interesse da primeira sessão para a segunda, ao lado da explicação sobre o curso.

### 2.2. Métricas

#### 2.2.1 Métrica primária

A principal métrica para avaliar o sucesso do teste é a taxa de conversão dos leads. Os dados são extraídos manualmente do HubSpot pela equipe da Wizard e enviados ao orientador da turma, que repassa os novos resultados aos alunos diariamente.

#### 2.2.2 Métricas secundárias

A métrica secundária é a taxa de cliques no botão do WhatsApp e Participe. As matrículas dos alunos também podem ser realizadas por esse canal, mas não receberemos imediatamente o número de leads gerados por ele. Portanto, continuaremos monitorando a quantidade de cliques no botão.

### 2.3 Hipóteses Estatísticas

#### Primeira Proposta

### Hipótese
Usuários não percebem o efeito no botão "Participe".

### Justificativa da Hipótese
Ao clicar no botão "Participe", o formulário é exibido imediatamente na primeira seção da página, o que pode causar confusão ou a impressão de que o botão não funciona, pois o formulário já está visível.

### Proposta B do Teste A/B
Ao deslocar o formulário para a segunda sessão da página, os usuários irão perceber melhor o efeito do botão “Participe” e farão mais inscrições pelo formulário

### Validação da Hipótese
A hipótese será validada se mover o formulário resultar em um aumento no número de cliques no botão "Participe" e nas inscrições através do formulário.

### Variáveis

**Variável Independente:**  
A posição do formulário de inscrição (primeira seção vs. segunda seção da página).

**Variável Dependente:**  
Número de cliques no botão "Participe" e número de inscrições no curso através do formulário.

### Grupos

**Grupo Experimental:**  
Usuários que acessam a versão da página onde o formulário está na segunda seção.

**Grupo de Controle:**  
Usuários que acessam a versão da página onde o formulário permanece na primeira seção.

#### Segunda Proposta

### Hipótese
Usuários não se inscrevem no formulário pois não possuem informações suficientes sobre o curso.

### Justificativa da Hipótese
Os usuários podem hesitar em preencher e enviar o formulário de inscrição porque sentem que não possuem informações detalhadas sobre o curso. Sem essas informações, eles podem se sentir inseguros ou não suficientemente motivados para completar a inscrição.

### Proposta B do Teste A/B

Ao deslocarmos o conteúdo explicativo do curso para o lado do formulário, os usuários irão se inscrever mais.

### Validação da Hipótese
A hipótese será validada se a adição do conteúdo explicativo do curso ao lado do formulário resultar em um aumento no número de envios de formulários.

### Variáveis

**Variável Independente:**  
A presença de conteúdo explicativo do curso ao lado do formulário (sim ou não).

**Variável Dependente:**  
Número de envios de formulários de inscrição.

### Grupos

**Grupo Experimental:**  
Usuários que acessam a versão da página onde o conteúdo explicativo do curso está ao lado do formulário (6% dos usuários).

**Grupo de Controle:**  
Usuários que acessam a versão da página onde o formulário está sem o conteúdo explicativo ao lado (94% dos usuários).

### 2.4. Período do Teste

A página experimental estará no ar de 04 de junho de 2024 a 14 de junho de 2024. Os dados sobre a evolução do teste serão recebidos diariamente e poderão ser extraídos tanto pelo HubSpot quanto pelo Looker.

## 3. Teste A/B


As mudanças foram realizadas na página de desenvolvimento da Wizard, acessível através de um link junto com o código-fonte, que podia ser acessado por meio de um tutorial. As principais alterações foram feitas em dois arquivos, "home" e "formulário", modificando o HTML e o CSS da página.

Após as configurações e a reestruturação da Proposta B, a página foi implantada no ambiente de homologação da Wizard e, no dia 04 de junho, foi lançada ao ar.

Para mais detalhes sobre o processo, acesse a seguinte documentação: [Código - Proposta B para Teste A/B.](https://github.com/Inteli-College/2024-1B-T04-SI10-G03/blob/main/document/sprint%203/%5BDocumenta%C3%A7%C3%A3o%20C%C3%B3digo%5D%20Proposta%20B%20para%20Teste%20A_B.pdf)

## 4. Preparação dos dados

Os resultados do teste serão disponibilizados em duas plataformas: HubSpot e Looker.

Até o momento, foram disponibilizados apenas dados relacionados a quantidade de sessões e a quantidade de leads. Acredita-se que, ao longo da próxima sprint, a Wizard fornecerá mais dados para complementar a análise.

### 4.1. Descrição dos Dados


- **`grupo`**: A qual grupo aquela linha de dados se refere.
- **`data_inicial`**: O range de data inicial que os dados foram coletados.
- **`data_final`**: O range de data final que os dados foram coletados.
- **`sessions`**: Número total de sessões no site.
- **`numero_leads`**: Quantidade de submissões de formulários.



### 4.2. Carregamento dos dados

In [ ]:
import pandas as pd

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
df_looker = pd.read_csv('/content/drive/Shareddrives/M10_SI/dados-analise-quantitativa-m10/looker - 11_06.csv')

In [ ]:
df_hubspot = pd.read_csv('/content/drive/Shareddrives/M10_SI/dados-analise-quantitativa-m10/Hubspot - 11_06.csv')

### 4.3. Informações dos dados

In [ ]:
df_looker.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6 entries, 0 to 5
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Pagina        6 non-null      object
 1   Sessoes       6 non-null      int64 
 2   Data inicial  6 non-null      object
 3   Data final    6 non-null      object
dtypes: int64(1), object(3)
memory usage: 320.0+ bytes


In [ ]:
df_hubspot.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6 entries, 0 to 5
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Pagina        6 non-null      object
 1   Conversoes    6 non-null      int64 
 2   Data inicial  6 non-null      object
 3   Data final    6 non-null      object
dtypes: int64(1), object(3)
memory usage: 320.0+ bytes


### 4.4. Visualização dos dados

In [ ]:
df_looker

Pagina  Sessoes Data inicial  Data final
0           Controle     6491   04/06/2024  11/06/2024
1  Teste B - Grupo 1      460   04/06/2024  11/06/2024
2  Teste B - Grupo 2      303   04/06/2024  11/06/2024
3  Teste B - Grupo 3      443   04/06/2024  11/06/2024
4  Teste B - Grupo 4      309   04/06/2024  11/06/2024
5  Teste B - Grupo 5      278   04/06/2024  11/06/2024

In [ ]:
df_hubspot

Pagina  Conversoes Data inicial  Data final
0           Controle         541   04/06/2024  11/06/2024
1  Teste B - Grupo 1          28   04/06/2024  11/06/2024
2  Teste B - Grupo 2          19   04/06/2024  11/06/2024
3  Teste B - Grupo 3          26   04/06/2024  11/06/2024
4  Teste B - Grupo 4          19   04/06/2024  11/06/2024
5  Teste B - Grupo 5          15   04/06/2024  11/06/2024

### 4.5. Unificando os dataframes

In [ ]:
dados = pd.concat([df_looker, df_hubspot['Conversoes']], axis=1)
dados

Pagina  Sessoes Data inicial  Data final  Conversoes
0           Controle     6491   04/06/2024  11/06/2024         541
1  Teste B - Grupo 1      460   04/06/2024  11/06/2024          28
2  Teste B - Grupo 2      303   04/06/2024  11/06/2024          19
3  Teste B - Grupo 3      443   04/06/2024  11/06/2024          26
4  Teste B - Grupo 4      309   04/06/2024  11/06/2024          19
5  Teste B - Grupo 5      278   04/06/2024  11/06/2024          15

### 4.6. Calculando a taxa de conversão

In [ ]:
dados['taxa_conversao'] = ((dados['Conversoes'] / dados['Sessoes']) * 100).round(2)
dados

Pagina  Sessoes Data inicial  Data final  Conversoes  \
0           Controle     6491   04/06/2024  11/06/2024         541   
1  Teste B - Grupo 1      460   04/06/2024  11/06/2024          28   
2  Teste B - Grupo 2      303   04/06/2024  11/06/2024          19   
3  Teste B - Grupo 3      443   04/06/2024  11/06/2024          26   
4  Teste B - Grupo 4      309   04/06/2024  11/06/2024          19   
5  Teste B - Grupo 5      278   04/06/2024  11/06/2024          15   

   taxa_conversao  
0            8.33  
1            6.09  
2            6.27  
3            5.87  
4            6.15  
5            5.40

## 5. Descrição dos dados

In [ ]:
# Estatísticas descritivas para sessões
media_sessoes = dados['Sessoes'].mean()
desvio_padrao_sessoes = dados['Sessoes'].std()
mediana_sessoes = dados['Sessoes'].median()
min_sessoes = dados['Sessoes'].min()
max_sessoes = dados['Sessoes'].max()

# Estatísticas descritivas para conversões
media_conversoes = dados['Conversoes'].mean()
desvio_padrao_conversoes = dados['Conversoes'].std()
mediana_conversoes = dados['Conversoes'].median()
min_conversoes = dados['Conversoes'].min()
max_conversoes = dados['Conversoes'].max()

# Estatísticas descritivas para taxa de conversão
media_taxa_conversao = dados['taxa_conversao'].mean()
desvio_padrao_taxa_conversao = dados['taxa_conversao'].std()
mediana_taxa_conversao = dados['taxa_conversao'].median()
min_taxa_conversao = dados['taxa_conversao'].min()
max_taxa_conversao = dados['taxa_conversao'].max()

# Imprimir as estatísticas descritivas
print("Estatísticas Descritivas para Sessões:")
print(f"Média: {media_sessoes}")
print(f"Desvio Padrão: {desvio_padrao_sessoes}")
print(f"Mediana: {mediana_sessoes}")
print(f"Min: {min_sessoes}")
print(f"Max: {max_sessoes}")

print("\nEstatísticas Descritivas para Conversões:")
print(f"Média: {media_conversoes}")
print(f"Desvio Padrão: {desvio_padrao_conversoes}")
print(f"Mediana: {mediana_conversoes}")
print(f"Min: {min_conversoes}")
print(f"Max: {max_conversoes}")

print("\nEstatísticas Descritivas para Taxa de Conversão:")
print(f"Média: {media_taxa_conversao}%")
print(f"Desvio Padrão: {desvio_padrao_taxa_conversao}%")
print(f"Mediana: {mediana_taxa_conversao}%")
print(f"Min: {min_taxa_conversao}%")
print(f"Max: {max_taxa_conversao}%")

Estatísticas Descritivas para Sessões:
Média: 1380.6666666666667
Desvio Padrão: 2504.7180014258424
Mediana: 376.0
Min: 278
Max: 6491

Estatísticas Descritivas para Conversões:
Média: 108.0
Desvio Padrão: 212.1810547622007
Mediana: 22.5
Min: 15
Max: 541

Estatísticas Descritivas para Taxa de Conversão:
Média: 6.351666666666667%
Desvio Padrão: 1.0165907075449128%
Mediana: 6.12%
Min: 5.4%
Max: 8.33%


## 6. Testes estatísticos

### 6.1 Teste t

**Hipótese nula (𝐻0):** A alteração na página (proposta B) não teve um impacto significativo na taxa de conversão (as taxas de conversão dos grupos de controle e experimental são iguais).

**Hipótese alternativa (𝐻𝐴):** A alteração na página (proposta B) teve um impacto significativo na taxa de conversão (as taxas de conversão dos grupos de controle e experimental são diferentes).

A escolha de utilizar o teste t para proporções em vez do teste t convencional se deve à natureza dos dados de conversão. Em testes A/B que avaliam taxas de conversão, estamos lidando com proporções, onde cada conversão é um evento binário (sucesso ou falha) e as taxas de conversão representam a proporção de sucessos em relação ao total de tentativas (sessões).

O teste t convencional é mais adequado para comparar médias de dados contínuos, enquanto o teste t de proporções leva em conta a natureza binária das conversões e a variabilidade específica das proporções, proporcionando uma análise mais precisa e relevante para este tipo de dados.

Portanto, o uso do teste t de proporções permite uma avaliação mais adequada da significância estatística das diferenças observadas nas taxas de conversão entre os grupos de controle e experimental.

In [ ]:
import pandas as pd
import numpy as np
from scipy import stats

def teste_t_proporcao(df, grupo_controle, grupo_teste):
    controle = df[df['Pagina'] == grupo_controle]
    n1 = controle['Sessoes'].values[0]
    x1 = controle['Conversoes'].values[0]

    teste = df[df['Pagina'] == grupo_teste]
    n2 = teste['Sessoes'].values[0]
    x2 = teste['Conversoes'].values[0]

    p1 = x1 / n1
    p2 = x2 / n2

    p_combinada = (x1 + x2) / (n1 + n2)

    denominador = np.sqrt(p_combinada * (1 - p_combinada) * (1/n1 + 1/n2))

    t_stat = (p1 - p2) / denominador

    df = min(n1, n2) - 1

    p_valor = 2 * (1 - stats.t.cdf(np.abs(t_stat), df))

    return t_stat, p_valor

grupo_controle = 'Controle'
grupo_teste = 'Teste B - Grupo 3'

t_stat, p_valor = teste_t_proporcao(dados, grupo_controle, grupo_teste)
print(f"Estatística t: {t_stat:.4f}")
print(f"Valor p: {p_valor:.4f}")

alpha = 0.05
if p_valor < alpha:
    print('Rejeitamos a hipótese nula (as taxas de conversão dos grupos de controle e experimental são significativamente diferentes).')
else:
    print('Não rejeitamos a hipótese nula (não há diferença significativa nas taxas de conversão entre os grupos de controle e experimental).')


Estatística t: 1.8323
Valor p: 0.0676
Não rejeitamos a hipótese nula (não há diferença significativa nas taxas de conversão entre os grupos de controle e experimental).


### 6.2 Intervalo de confiança

O intervalo de confiança é uma medida que estima a precisão das taxas de conversão observadas nos grupos de controle e experimental. Ele fornece uma faixa de valores que provavelmente contém a verdadeira taxa de conversão populacional, considerando a variação amostral. Para este relatório, utilizamos um intervalo de confiança de 95%.

In [ ]:
import pandas as pd
import numpy as np
from scipy import stats

# Função para calcular intervalo de confiança para proporções
def calc_intervalo_confianca(p, n, alpha=0.05):
    z = stats.norm.ppf(1 - alpha/2)
    margin_of_error = z * np.sqrt((p * (1 - p)) / n)
    return (p - margin_of_error, p + margin_of_error)

# Recriar o DataFrame a partir da imagem fornecida
dados = pd.DataFrame({
    'Pagina': ['Controle', 'Teste B - Grupo 1', 'Teste B - Grupo 2', 'Teste B - Grupo 3', 'Teste B - Grupo 4', 'Teste B - Grupo 5'],
    'Sessoes': [6491, 460, 303, 443, 309, 278],
    'Data_inicial': ['04/06/2024']*6,
    'Data_final': ['11/06/2024']*6,
    'Conversoes': [541, 28, 19, 26, 19, 15],
    'taxa_conversao': [8.33, 6.09, 6.27, 5.87, 6.15, 5.40]
})

# Filtrar para os grupos de interesse
dados_interesse = dados[dados['Pagina'].isin(['Controle', 'Teste B - Grupo 3'])]

# Calculando intervalo de confiança para o grupo controle e o grupo 3
intervalos_confianca = []
for index, row in dados_interesse.iterrows():
    p = row['taxa_conversao'] / 100
    n = row['Sessoes']
    intervalo = calc_intervalo_confianca(p, n)
    intervalos_confianca.append(intervalo)

dados_interesse['intervalo_confianca'] = intervalos_confianca

# Formatando intervalos com menos casas decimais
dados_interesse['intervalo_confianca'] = dados_interesse['intervalo_confianca'].apply(lambda x: (round(x[0]*100, 2), round(x[1]*100, 2)))

print(dados_interesse[['Pagina', 'taxa_conversao', 'intervalo_confianca']])


              Pagina  taxa_conversao intervalo_confianca
0           Controle            8.33         (7.66, 9.0)
3  Teste B - Grupo 3            5.87        (3.68, 8.06)


<ipython-input-63-e02a6cb3e85b>:32: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

<ipython-input-63-e02a6cb3e85b>:35: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



Os intervalos de confiança ajudam a entender a precisão das taxas de conversão estimadas e permitem comparar se as diferenças observadas entre os grupos de controle e experimental são estatisticamente significativas.

Resultados:

* **Grupo de Controle:** A taxa de conversão é de 8.33%, com um intervalo de confiança de [7.71%, 8.95%].

* **Grupo Experimental 3:** A taxa de conversão é de 5.87%, com um intervalo de confiança de [4.14%, 7.60%].

### 6.3 Teste de Fisher

O Teste de Fisher é utilizado para determinar se há uma associação significativa entre duas variáveis categóricas em uma tabela de contingência. Ele é especialmente útil quando as amostras são pequenas. No contexto deste teste A/B, o Teste de Fisher foi aplicado para avaliar se a distribuição de conversões difere entre o grupo de controle e o grupo experimental 3.

**Hipóteses:**

* **Hipótese Nula (H0):** Não há associação entre o grupo (controle ou experimental) e a taxa de conversão (as proporções de conversões são iguais entre os grupos).
* **Hipótese Alternativa (HA):** Existe uma associação entre o grupo (controle ou experimental) e a taxa de conversão (as proporções de conversões são diferentes entre os grupos).

In [ ]:
import pandas as pd
import numpy as np
from scipy import stats

def calc_intervalo_confianca(p, n, alpha=0.05):
    z = stats.norm.ppf(1 - alpha/2)
    margin_of_error = z * np.sqrt((p * (1 - p)) / n)
    return (p - margin_of_error, p + margin_of_error)

dados_interesse = dados[dados['Pagina'].isin(['Controle', 'Teste B - Grupo 3'])].copy()

intervalos_confianca = []
for index, row in dados_interesse.iterrows():
    p = row['taxa_conversao'] / 100
    n = row['Sessoes']
    intervalo = calc_intervalo_confianca(p, n)
    intervalos_confianca.append(intervalo)

dados_interesse.loc[:, 'intervalo_confianca'] = intervalos_confianca

dados_interesse.loc[:, 'intervalo_confianca'] = dados_interesse['intervalo_confianca'].apply(lambda x: (round(x[0]*100, 2), round(x[1]*100, 2)))

print(dados_interesse[['Pagina', 'taxa_conversao', 'intervalo_confianca']])

tabela_fisher = pd.DataFrame({
    'Conversoes': [dados_interesse[dados_interesse['Pagina'] == 'Controle']['Conversoes'].values[0],
                   dados_interesse[dados_interesse['Pagina'] == 'Teste B - Grupo 3']['Conversoes'].values[0]],
    'Nao_Conversoes': [dados_interesse[dados_interesse['Pagina'] == 'Controle']['Sessoes'].values[0] - dados_interesse[dados_interesse['Pagina'] == 'Controle']['Conversoes'].values[0],
                       dados_interesse[dados_interesse['Pagina'] == 'Teste B - Grupo 3']['Sessoes'].values[0] - dados_interesse[dados_interesse['Pagina'] == 'Teste B - Grupo 3']['Conversoes'].values[0]]
}, index=['Controle', 'Teste B - Grupo 3'])

oddsratio, p_value = stats.fisher_exact(tabela_fisher)
print(f"Razão de Chances: {oddsratio:.4f}")
print(f"Valor p: {p_value:.4f}")


              Pagina  taxa_conversao intervalo_confianca
0           Controle            8.33         (7.66, 9.0)
3  Teste B - Grupo 3            5.87        (3.68, 8.06)
Razão de Chances: 1.4583
Valor p: 0.0726


Resultados:

* *Grupo de controle:* A taxa de conversão é de 8.33%, com um intervalo de confiança de [7.66%, 9.00%].

* *Grupo experimental 3:* A taxa de conversão é de 5.87%, com um intervalo de confiança de [3.68%, 8.06%].

* *Razão de chances:* 1.4583, indicando que a chance de conversão no grupo experimental é aproximadamente 1.46 vezes a do grupo de controle.

* *Valor p*: 0.0726. Como o valor p é maior que 0.05, não rejeitamos a hipótese nula. Isso sugere que não há evidência suficiente para afirmar que as proporções de conversões diferem significativamente entre os grupos de controle e experimental.

Os resultados do Teste de Fisher indicam que, embora a razão de chances sugira que a chance de conversão no grupo experimental seja 1.46 vezes a do grupo de controle, o valor p (0.0726) não é suficientemente baixo para rejeitar a hipótese nula. Portanto, não há evidência estatística suficiente para afirmar que as proporções de conversões diferem significativamente entre os grupos de controle e experimental.

## 7. Análise dos resultados

### 7.1 Visualização dos resultados

In [ ]:
import plotly.graph_objects as go
from plotly.subplots import make_subplots

fig = make_subplots(rows=3, cols=1, shared_xaxes=True,
                    vertical_spacing=0.1, subplot_titles=("Sessões por Página", "Conversões por Página", "Taxa de Conversão por Página"))

fig.add_trace(go.Scatter(x=dados['Pagina'], y=dados['Sessoes'], mode='lines+markers', name='Sessões'), row=1, col=1)

fig.add_trace(go.Scatter(x=dados['Pagina'], y=dados['Conversoes'], mode='lines+markers', name='Conversões'), row=2, col=1)

fig.add_trace(go.Scatter(x=dados['Pagina'], y=dados['taxa_conversao'], mode='lines+markers', name='Taxa de Conversão (%)'), row=3, col=1)

fig.update_layout(height=800, width=800, title_text="Resultados do Teste A/B")

fig.show()

### 7.2 Intepretação dos resultados

#### **Resultados do Teste t**

* ***t-statistic:*** *1.8323*
* ***p-value:*** *0.0676*

**1. t-statistic:** Este valor indica a magnitude da diferença entre as proporções das taxas de conversão dos grupos de controle e experimental em unidades de erro padrão. Um valor de 1.83 sugere que há uma diferença moderada entre as proporções das taxas de conversão.

**2. p-value:** Este valor indica a probabilidade de observar uma diferença tão grande (ou maior) entre as proporções das taxas de conversão dos grupos de controle e experimental se a hipótese nula fosse verdadeira (isto é, se não houvesse diferença real entre as proporções).


Como o p-value (0.0676) é maior que o nível de significância (α = 0.05), não rejeitamos a hipótese nula. Isso significa que não há evidências estatísticas de que as taxas de conversão dos grupos de controle e experimental são diferentes. Em outras palavras, a alteração na página (proposta B do teste A/B) não resultou em uma diferença significativa na taxa de conversão em comparação com o controle. Portanto, qualquer diferença observada pode ser atribuída ao acaso.



#### **Intervalo de confiança**

* **Grupo de controle**: A taxa de conversão é de 8.33%, com um intervalo de confiança de [7.66%, 9.00%].

* **Grupo Experimental 3:** A taxa de conversão é de 5.87%, com um intervalo de confiança de [3.68%, 8.06%].

Os intervalos de confiança mostram que, embora a taxa de conversão do grupo experimental 3 seja menor que a do grupo de controle, as faixas de confiança se sobrepõem consideravelmente. Isso indica que a variação observada nas taxas de conversão pode ser atribuída à variação amostral, e não há uma diferença estatisticamente significativa entre os grupos.

#### **Resultados do Teste de Fisher**

**Razão de chances:** 1.4583, indicando que a chance de conversão no grupo experimental é aproximadamente 1.46 vezes a do grupo de controle.

**Valor p**: 0.0726

O Teste de Fisher reforça a conclusão de que não há uma diferença estatisticamente significativa entre as taxas de conversão dos grupos de controle e experimental. Embora a razão de chances sugira uma maior probabilidade de conversão no grupo experimental, o p-value de 0.0726 é maior que o nível de significância de 0.05, indicando que a diferença observada não é estatisticamente significativa.

#### **Conclusão da interpretação:**

Os resultados dos testes estatísticos - incluindo o teste t e o Teste de Fisher - demonstram que as alterações implementadas na página do grupo experimental (proposta B) não resultaram em uma diferença significativa na taxa de conversão em comparação com o grupo de controle. Apesar das melhorias visuais e de navegação, como o reposicionamento do formulário, a adição de ícones representativos e a ampliação da imagem promocional, as mudanças não tiveram o impacto esperado na conversão de visitantes em leads.

Portanto, a análise sugere que as modificações não proporcionaram um aumento significativo na taxa de conversão e que qualquer diferença observada pode ser atribuída ao acaso. É recomendável revisar as mudanças propostas e considerar outras abordagens ou melhorias para atingir os objetivos desejados de aumento da taxa de conversão.

## 8. Limitações

O teste A/B realizado para a otimização da página da Wizard apresentou diversas limitações que precisam ser cuidadosamente consideradas para uma avaliação precisa dos resultados. Primeiramente, o período de teste foi relativamente curto, durando apenas uma semana. Essa brevidade pode não ter sido suficiente para capturar uma amostra representativa e consistente do comportamento dos usuários. Em períodos tão curtos, flutuações sazonais ou eventos específicos podem influenciar os resultados, comprometendo a generalização dos achados.


Outra limitação significativa foi a pequena amostra de usuários exposta às alterações na página B. A proporção limitada de participantes no grupo experimental afetou a significância estatística dos resultados, dificultando a detecção de diferenças significativas entre os grupos de controle e experimental. Com uma amostra reduzida, é mais difícil obter resultados conclusivos sobre a eficácia das mudanças implementadas.

Além disso, foram implementadas diversas mudanças simultâneas na página B, incluindo o reposicionamento do formulário, a adição de ícones representativos e a ampliação da imagem promocional. Embora essas alterações visassem melhorar a usabilidade e a estética, a multiplicidade de mudanças dificultou a identificação de quais modificações específicas tiveram impacto nas taxas de conversão. Essa abordagem múltipla impede uma análise clara sobre qual mudança foi realmente eficaz.

Outro aspecto a ser considerado são as variações no contexto de uso, como diferentes dispositivos, horários de acesso e perfis de usuário. Estas variações podem influenciar significativamente a interação dos usuários com a página, e o teste realizado não levou em conta os dados obtidos a partir  dessas segmentações adicionais, tendo em vista que a Pearson não estruturou o Google Analytics para capturá-los nas páginas B's. Isso pode limitar a compreensão do comportamento dos usuários e das necessidades específicas de diferentes segmentos, além, é claro, do impacto da outra versão da página na usabilidade dos usuários.


Por fim, a análise qualitativa não foi realizada, visto que todo esse relatório foi feito focando, principalmente, em métricas quantitativas, sem uma investigação profunda do feedback qualitativo dos usuários. Comentários e percepções dos usuários podem oferecer insights valiosos sobre a experiência e possíveis pontos de melhoria que não são capturados pelas métricas quantitativas.

Portanto, as principais limitações do teste A/B realizado incluem o período de teste curto, a pequena amostra de usuários no grupo experimental, a coleta de dados manual, a implementação de múltiplas mudanças simultâneas, a falta de segmentação no contexto de uso e a análise qualitativa limitada. Essas limitações devem ser levadas em consideração ao interpretar os resultados e ao planejar futuras otimizações.

## 9 Conclusão

### 9.1 Resumo dos achados

O teste A/B realizado para avaliar as alterações na página da Wizard teve como objetivo principal aumentar a taxa de conversão de leads em 7% e melhorar a qualidade dos leads capturados. Foram implementadas diversas modificações na página, incluindo o reposicionamento do formulário de inscrição para a segunda seção, a adição de ícones representativos dos produtos oferecidos e a ampliação da imagem promocional.

No entanto, os resultados obtidos indicaram que essas mudanças não tiveram o impacto esperado. A análise estatística revelou uma estatística t de 1.8323 e um valor p de 0.0676, o que significa que não podemos rejeitar a hipótese nula. Portanto, não há evidências estatísticas de que as taxas de conversão dos grupos de controle e experimental são significativamente diferentes. Especificamente, a taxa de conversão do grupo experimental 3 foi de 5.87%, enquanto a do grupo de controle foi de 8.33%. Esses dados reforçam a conclusão de que as modificações implementadas não resultaram em um aumento significativo na taxa de conversão.

O Teste de Fisher também corroborou esses resultados, com uma razão de chances de 1.4583 e um valor p de 0.0726, indicando que a diferença observada não é estatisticamente significativa. Embora a chance de conversão no grupo experimental pareça ser maior, a variação pode ser atribuída ao acaso.

### 9.2 Recomendações

Diante dos resultados obtidos, é essencial adotar uma abordagem mais detalhada e criteriosa para futuras otimizações da página. Abaixo estão algumas recomendações específicas para a Pearson continuar implementando e aprimorando o teste A/B:

1. **Extensão do período de teste:**
   O período de uma semana é insuficiente para capturar uma amostra representativa do comportamento dos usuários. Recomenda-se estender o período de teste para pelo menos quatro semanas. Isso permitirá captar variações sazonais e comportamentais mais amplas, oferecendo dados mais robustos e representativos.

2. **Aumento da amostra experimental:**
   A pequena amostra de usuários no grupo experimental limitou a significância estatística dos resultados. Sugerimos aumentar a proporção de usuários expostos às alterações na página. Uma divisão de 50% para cada grupo (controle e experimental) pode proporcionar um equilíbrio melhor para detectar diferenças significativas. Alternativamente, garantir que pelo menos 20% dos visitantes sejam alocados ao grupo experimental pode ajudar a obter dados mais significativos.

3. **Automatização da coleta de dados:**
   A coleta manual de dados pode introduzir erros e inconsistências. Implementar sistemas automatizados para coleta e análise de dados, como Google Analytics, Mixpanel ou HubSpot, pode reduzir o risco de erros e aumentar a precisão dos insights derivados dos dados.

4. **Abordagem iterativa de mudanças:**
   Testar múltiplas mudanças simultaneamente dificultou a identificação do impacto individual de cada alteração. Recomenda-se adotar uma abordagem iterativa, testando uma mudança por vez. Isso permitirá isolar e identificar o efeito de cada alteração individualmente, começando com aquelas que teoricamente têm maior impacto na experiência do usuário e na taxa de conversão.

5. **Segmentação adicional dos testes:**
   Considerar variações no contexto de uso, como diferentes dispositivos (desktop vs. mobile), horários de acesso (diurno vs. noturno) e perfis de usuário. Realizar segmentações adicionais permitirá uma compreensão mais aprofundada do comportamento de diferentes grupos de usuários e adaptar a página para otimizar a conversão para cada segmento específico.

6. **Integração de feedback qualitativo:**
   Além das métricas quantitativas, é crucial coletar feedback qualitativo dos usuários. Utilizar ferramentas de feedback, como pesquisas de satisfação e testes de usabilidade, pode oferecer insights valiosos sobre a experiência do usuário que não são capturados pelas métricas quantitativas. Isso permitirá identificar áreas de melhoria e ajustar a página de acordo com as necessidades e preferências dos usuários.

7. **Monitoramento contínuo e iterativo:**
   Após implementar as mudanças recomendadas, é importante continuar monitorando e ajustando a página com base nos dados coletados. Um processo contínuo de teste e otimização assegura que a página permaneça eficaz e alinhada com as expectativas dos usuários. Realizar testes de acompanhamento para verificar se as mudanças estão mantendo o impacto positivo ao longo do tempo.

Implementando essas recomendações, a Pearson poderá continuar otimizando a landing page de forma mais eficaz, melhorando a experiência do usuário e potencialmente aumentando as taxas de conversão de maneira sustentável. Essas ações não apenas contribuirão para alcançar os objetivos estabelecidos, mas também fortalecerão a capacidade da Wizard de competir no mercado educacional online, garantindo que a oferta de cursos atenda e supere as expectativas dos alunos.